numpy/core/src/umath/simd.inc.src

```diff
@@ -33,6 +33,7 @@
 #include <string.h> /* for memcpy */
 
 #define VECTOR_SIZE_BYTES 16
+#define MAX_STEP_SIZE 2097152
```

Comment:
Can you explain where this came from? Are you looking for `NPY_MAX_INT32 / 16`, where 16 is the maximum number of lanes?

Reply:
I don't have a satisfactory explanation to this. What I ended up using is max step size = 2MB since it satisfies 2 criteria:

1) Typical linux kernel page size is 4Kb, but sometimes it could also be 2MB which is == 2097152 Bytes. For a step size as large as this, surely all the 16 loads falls on 16 different pages which I would think is the worst performance scenario. 
2) And it additionally satisfies `MAX_STEP_SIZE*16/esize < NPY_MAX_INT32 ` which allows us to use `i32gather/scatter` without problems.

Reply:
Your comment above should go in the source code.